datafusion/core/src/dataframe/mod.rs

```diff
@@ -1309,24 +1314,25 @@ impl DataFrame {
             Column::from_qualified_name_ignore_case(old_name)
         };
 
-        let field_to_rename = match self.plan.schema().field_from_column(&old_column) {
-            Ok(field) => field,
-            // no-op if field not found
-            Err(DataFusionError::SchemaError(SchemaError::FieldNotFound { .. }, _)) => {
-                return Ok(self)
-            }
-            Err(err) => return Err(err),
-        };
+        let (qualifier_rename, field_rename) =
+            match self.plan.schema().qualified_field_from_column(&old_column) {
```

Comment:
I think it is possible to avoid clone for `Field` here since we can do comparison with `&Field`

Reply:
It is nice to minimize the clone, especially for basic types that may be called many times and many places e.g. DFSchema